core/src/components/input-password-toggle/input-password-toggle.tsx

```diff
@@ -0,0 +1,151 @@
+import type { ComponentInterface } from '@stencil/core';
+import { Component, Element, Host, Prop, h, Watch } from '@stencil/core';
+import { printIonWarning } from '@utils/logging';
+import { createColorClasses } from '@utils/theme';
+import { eyeOff, eye } from 'ionicons/icons';
+
+import { getIonMode } from '../../global/ionic-global';
+import type { Color, TextFieldTypes } from '../../interface';
+
+/**
+ * @virtualProp {"ios" | "md"} mode - The mode determines which platform styles to use.
+ */
+@Component({
+  tag: 'ion-input-password-toggle',
+  styleUrls: {
+    ios: 'input-password-toggle.ios.scss',
+    md: 'input-password-toggle.md.scss',
+  },
```

Comment:
I need to look closer at this tomorrow, but without these stylesheets inheriting the mode form `ion-input-password-toggle` to the inner `ion-button does not work: The mode is always "MD". This shouldn't block a review, but I did add this back for now so we can get the build passing.